pkg/scheduler/framework/types.go

```diff
@@ -224,13 +224,20 @@ type FitError struct {
 	Pod         *v1.Pod
 	NumAllNodes int
 	Diagnosis   Diagnosis
+	// CachedMsg is calculated only once.
+	// Protect using sync.Once is unnecessary as FitError is read in serial.
+	CachedMsg string
```

Comment:
this is awkward, can we just make sure that Error is called once?

Reply:
It's hard to control because Error's evaluation can be implicit, like klog in [MakeDefaultErrorFunc](https://github.com/kubernetes/kubernetes/blob/5583f0c0659b4033869939ad4c8ee3bca156386b/pkg/scheduler/scheduler.go#L360). Also, out-of-tree plugins may also use klog to dump error arbitrarily.

Reply:
> It's hard to control because Error's evaluation can be implicit,

Still find strange that we would get the error message in more than one place.

> Also, out-of-tree plugins may also use klog to dump error arbitrarily.

Where would out of tree plugins get this error?